README.md

```diff
@@ -2,622 +2,566 @@
+### Errors
 
-By default, `o-forms` listens to the `blur` event of an input field, when a user leaves a field, the input will be validated and if an invalid input is found, the error class will be added to the input.
+In trms of invalid inputs, `o-forms` has a build in mechanism to display custom errors and an error summary. If neither of these work for your product, you can also default to [native browser validation](#form-instance).
 
-Additionally `o-forms` fires an event `oForms.toggled` when a toggle checkbox is clicked.
-
-#### Constructing
-
-An o-forms object must be constructed for every `<form>` element you have on your page that you want to validate with this module. You can do this for explicit elements like so:
-
-```js
-const OForms = require('o-forms');
-const formsEl = document.querySelector('#form-element');
-
-const forms = new OForms(formsEl);
-```
-
-Alternatively, an o.DOMContentLoaded event can be dispatched on the document to auto-construct an o-forms object for each element with a `data-o-component="o-forms"` attribute:
-
-```js
-require('o-forms');
-document.addEventListener("DOMContentLoaded", function() {
-    document.dispatchEvent(new CustomEvent('o.DOMContentLoaded'));
-});
+#### Custom Errors
+In order to provide customised error messages for an invalid input field, you'll need to add the message to the markup. In any of the structures shown in the markup above, the error message will need to live within the input container as a sibling to the input elements:
+```html
+...
+	<span class="o-forms-input">
+		<!-- inputs -->
+		<span class="o-forms-input__error">This is the error message</span>
+	</span>
+...
 ```
+The message is hidden by default, until the input field become invalid. 
 
-#### Validate on form submit
+#### Error Summary
+`o-forms` also generates an error message element when a form is submitted and invalid inputs are recognised.
 
-By default `o-forms` tests inputs on the `blur` event for each input. To defer the validation to the `submit` event of the form, you can pass an config object to set the `testEvent` when constructing `o-forms`:
+This feature will collect the custom messages of the invalid fields if they are present in the markup, or will default to the browsers native error message if they aren't. It will associate the message to the title of the input that is invalid, generate a list of links at the top of the form, and focus on the first link.
 
-```js
-const OForms = require('o-forms');
-const formsEl = document.querySelector('#form-element');
+This markup will always generated dynamically if the [errorSummary option](#form-instance) is set to `true`.
 
-const forms = new OForms(formsEl, { testEvent: 'submit' });
+## Sass
+You can include styles for every input type and feature by calling the `o-forms` primary mixin:
```

Comment:
I may be misunderstanding, but I think the primary mixin is called `oForms` and the sass file to import is called `o-forms`. 
I'm not sure if my suggestion is correct though so feel free to resolve this without changing it.
```suggestion
You can include styles for every input type and feature by calling the `oForms` primary mixin:
```

Reply:
Oh it was meant to imply the primary mixin that belongs to `o-forms` but yeah now that you point it out it is kind of confusing 🤔